### ISSUE TITLE
`identical_operands` incorrectly gets triggered in closure

### ISSUE THREAD

**Issue by michaeleustace** [realm/SwiftLint#3865]
### New Issue Checklist

- [x] Updated SwiftLint to the latest version
- [x] I searched for [existing GitHub issues](https://github.com/realm/SwiftLint/issues)

### Describe the bug

The following code gets reported as a violation of the `identical_operands` rule when the operands are from different objects:
```swift
enum MyType: Int {
    case one = 1
    case two = 2
    case three = 3
}

struct Wrapper {
    let type: MyType
}

// This triggers violation
[Wrapper(type: .three), Wrapper(type: .one), Wrapper(type: .two)].sorted { "\($0.type)" > "\($1.type)" }
```

##### Complete output when running SwiftLint, including the stack trace and command used

```bash
Pods/SwiftLint/swiftlint lint ./LintTest.swift 
Linting Swift files at paths ./LintTest.swift
Linting 'LintTest.swift' (1/1)
/Users/michaele/Projects/GitLab/myProject/LintTest.swift:27:62: warning: Identical Operands Violation: Comparing two identical operands is likely a mistake. (identical_operands)
Done linting! Found 1 violation, 0 serious in 1 file.
```

### Environment

* SwiftLint version : `0.46.3`
* Installation method used: Cocoapods
* Paste your configuration file:

```yml
whitelist_rules:

# CAN BE MADE ERRORS:
  - closing_brace
  - closure_end_indentation
  - closure_parameter_position
  - closure_spacing
  - comma
  - computed_accessors_order
  - contains_over_filter_count
  - contains_over_filter_is_empty
  - contains_over_first_not_nil
  - contains_over_range_nil_comparison
  - control_statement
  - empty_collection_literal
  - empty_count
  - empty_parameters
  - empty_parentheses_with_trailing_closure
  - empty_string
  - file_name_no_space
  - first_where
  - force_try
  - function_body_length
  - function_parameter_count
  - last_where
  - leading_whitespace
  - no_space_in_method_call
  - opening_brace
  - operator_usage_whitespace
  - operator_whitespace
  - private_action
  - redundant_nil_coalescing
  - redundant_void_return
  - return_arrow_whitespace
  - syntactic_sugar
  - toggle_bool
  - unneeded_parentheses_in_closure_argument
  - unused_capture_list
  - vertical_whitespace
  - vertical_whitespace_closing_braces
  - void_return
  
# NEED RECTIFYING AS WARNINGS:
  - file_length
  - force_cast
  - todo
  - expiring_todo
  - class_delegate_protocol
  - private_outlet
  - identical_operands
  - unused_closure_parameter
  - sorted_first_last
  - reduce_boolean
  - flatmap_over_map_reduce
  - for_where
  - overridden_super_call

excluded:
  - Pods

weak_delegate:
  severity: warning
  excluded:
    - ".*Test\\.swift"
    - ".*Tests\\.swift"

force_cast:
  severity: warning
  excluded:
    - ".*Test\\.swift"
    - ".*Tests\\.swift"

force_try:
  severity: warning
  excluded:
    - ".*Test\\.swift"
    - ".*Tests\\.swift"

```

* Are you using nested configurations? No
* Which Xcode version are you using (check `xcodebuild -version`)? `Xcode 13.2.1 Build version 13C100`
* Do you have a sample that shows the issue? Run `echo "[string here]" | swiftlint lint --no-cache --use-stdin --enable-all-rules`
  to quickly test if your example is really demonstrating the issue. If your example is more
  complex, you can use `swiftlint lint --path [file here] --no-cache --enable-all-rules`.

```swift
enum MyType: Int {
    case one = 1
    case two = 2
    case three = 3
}

struct Wrapper {
    let type: MyType
}

// This triggers violation
[Wrapper(type: .three), Wrapper(type: .one), Wrapper(type: .two)].sorted { "\($0.type)" > "\($1.type)" }
```

**Comment by marcelofabri** [comment]
Fixed by the SwiftSyntax rewrite
